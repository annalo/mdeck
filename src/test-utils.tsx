/* eslint-disable import/no-extraneous-dependencies */

import React from "react";
import { render } from "@testing-library/react";
import { renderHook } from "@testing-library/react-hooks";

import { ThemeProvider } from "components/Home/ThemeProvider";
import { MarkdownContextProvider } from "contexts/MarkdownContext";
import { SlideObserverProvider } from "contexts/SlideObserver";
import { CodeLineObserverProvider } from "contexts/CodeLineObserver";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider>
      <MarkdownContextProvider>
        <SlideObserverProvider>
          <CodeLineObserverProvider>{children}</CodeLineObserverProvider>
        </SlideObserverProvider>
      </MarkdownContextProvider>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const customRenderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

// override render and renderHook methods
export { customRender as render, customRenderHook as renderHook };
