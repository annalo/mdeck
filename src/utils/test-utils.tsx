/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import { render } from "@testing-library/react";
import { renderHook } from "@testing-library/react-hooks";

import { ThemeProvider } from "components/App/ThemeProvider";

import { MarkdownContextProvider } from "contexts/MarkdownContext";
import { PresentationContextProvider } from "contexts/PresentationContext";
import { SlideObserverProvider } from "contexts/SlideObserver";
import { CodeLineObserverProvider } from "contexts/CodeLineObserver";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider>
      <MarkdownContextProvider>
        <PresentationContextProvider>
          <SlideObserverProvider>
            <CodeLineObserverProvider>{children}</CodeLineObserverProvider>
          </SlideObserverProvider>
        </PresentationContextProvider>
      </MarkdownContextProvider>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const customRenderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

// override render and renderHook methods
export { customRender as render, customRenderHook as renderHook };
