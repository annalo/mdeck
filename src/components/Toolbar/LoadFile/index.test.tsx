import React from "react";
import { render } from "test-utils";
import { LoadFileMenuItem } from ".";

describe("<LoadFileMenuItem />", () => {
  test("should render and match the snapshot", () => {
    const { asFragment } = render(<LoadFileMenuItem />);
    expect(asFragment()).toMatchSnapshot();
  });
});
