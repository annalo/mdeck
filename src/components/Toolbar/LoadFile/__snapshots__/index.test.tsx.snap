// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadFileMenuItem /> should render and match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: none;
}

<input
    accept=".md,.txt"
    class="c0"
    id="fileLoader"
    title="Load Markdown File"
    type="file"
  />
  .c0 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 3px;
}

.c0 svg {
  fill: #d4d4d4;
}

.c0 svg circle,
.c0 svg path {
  stroke: #888888;
}

.c0 svg {
  -webkit-transition: 0.5s ease-in-out;
  transition: 0.5s ease-in-out;
}

.c0:hover svg {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  fill: #bababa;
}

.c0:hover svg circle,
.c0:hover svg path {
  stroke: #2f2f2f;
}

.c0:hover svg {
  -webkit-transition: opacity 0.4s ease-in-out;
  transition: opacity 0.4s ease-in-out;
  opacity: 0.3;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.c1 {
  position: absolute;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0;
  color: #2f2f2f;
  font-weight: bolder;
  font-size: 8px;
  -webkit-transition: 0.2s ease-in-out;
  transition: 0.2s ease-in-out;
}

.c1:hover {
  opacity: 1;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

<li
    class="c0"
  >
    <svgr-mock />
    <div
      class="c1"
    >
      LOAD
    </div>
  </li>
</DocumentFragment>
`;
